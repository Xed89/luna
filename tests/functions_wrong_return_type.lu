type Program
  fun main()

  fun getNumber(): int 
    return "7"