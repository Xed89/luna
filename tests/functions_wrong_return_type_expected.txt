Parsing completed
Syntax Tree:
TypeDeclaration 'Program'
  [0]: 
    Function 'main'
      type:
        null
      statements:
  [1]: 
    Function 'getNumber'
      type:
        Type 'int'
      statements:
        [0]: 
          ReturnStatementSyntax
            value:
              ExpressionLiteralSyntax '7'

Compile failed:
Function must return a value of type int, not string

